torch/_decomp/decompositions.py

```diff
@@ -1982,7 +1982,7 @@ def nll_loss_forward(
     elif ignore_index >= 0:
         total_weight = (target != ignore_index).sum().to(self)
     else:
-        total_weight = self.new_full((), 1.0 * result.numel())
+        total_weight = self.new_full((), result.numel())
```

Comment:
This looks wrong, the dtype is no longer float after this change

Reply:
yeah agreed the resultant dtype would be the same as that of `self`. It was triggering https://github.com/pytorch/pytorch/blob/c89d286af633a802226c34ccbdd5c7c4be10dcfb/torch/csrc/jit/python/init.cpp#L1403-L1407 which was causing issues because the output should be a `SymFloat`.